metadata-ingestion/src/datahub/ingestion/source/tableau/tableau.py

```diff
@@ -420,6 +420,17 @@ class TableauConfig(
         description="[advanced] Number of workbooks to query at a time using the Tableau API.",
     )
 
+    # Since the field upstream query was separated from the embedded datasource queries into an independent query,
+    # the number of queries increased significantly and so the execution time.
+    # To increase the batching and so reduce the number of queries, we can increase the page size for that
+    # particular case.
+    #
+    # `get_connection_objects_query_*` metrics in the report will help to understand the impact of this change.
+    embedded_datasource_field_upstream_page_size: int = Field(
+        default=100,
+        description="[advanced] Number of upstream fields to query at a time for embedded datasources using the Tableau API.",
+    )
```

Comment:
```suggestion
    datasource_field_page_size: int = Field(
        default=100,
        description="[advanced] Number of datasource fields to query at a time for upstreams using the Tableau API.",
    )
```
I believe this is applicable for published datasources as well